aws/config.go

```diff
@@ -300,10 +302,23 @@ func (c *Config) Client() (interface{}, error) {
 		opt.Config.Logger = awsLogger{}
 	}
 
-	if c.Insecure {
+	if c.Insecure || c.ProxyURL != "" {
+
 		transport := opt.Config.HTTPClient.Transport.(*http.Transport)
```

Comment:
This isn't part of this PR, but this is pretty fragile, we will probably need to rethink this down the road if we need to modify the transport in other ways or add a `RoundTripper`.